src/Orchard.Web/Modules/Orchard.OpenId/InMemoryCache.cs

```diff
@@ -22,51 +21,30 @@ public class InMemoryCache : TokenCache {
             Load();
         }
 
-        public static Dictionary<string, byte[]> InMemoryTokenCache {
+        public static ConcurrentDictionary<string, byte[]> InMemoryTokenCache {
```

Comment:
Does it have to be public? If not then remove it and put the initialization in the ctor